src/sp/transport/udp/udp.c

```diff
@@ -964,7 +965,17 @@ udp_pipe_recv(void *arg, nni_aio *aio)
 		nni_mtx_unlock(&ep->mtx);
 		return;
 	}
-
+	//If there are messages in the queue, retrieve the messages from the queue during reception 
+	//without waiting for the udp_recv_data operation to complete.
+	if(!nni_lmq_empty(&p->rx_mq)){
+		nni_lmq_get(&p->rx_mq, &msg);
+		nni_aio_set_msg(aio, msg);
+		nni_aio_completions_add(
+			&ep->complq, aio, 0, nni_aio_count(aio)
+		);
```

Comment:
Please move this to the line above... 
In fact, it might work out that this all can go on one line.  clang-format will automatically format for the width.